app/src/main/java/org/schabi/newpipe/player/playback/MediaSourceManager.java

```diff
@@ -422,21 +421,24 @@ private void maybeLoadItem(@NonNull final PlayQueueItem item) {
 
     private Single<ManagedMediaSource> getLoadedMediaSource(@NonNull final PlayQueueItem stream) {
         return stream.getStream().map(streamInfo -> {
-            final MediaSource source = playbackListener.sourceOf(stream, streamInfo);
-            if (source == null || !MediaItemTag.from(source.getMediaItem()).isPresent()) {
-                final String message = "Unable to resolve source from stream info. "
-                        + "URL: " + stream.getUrl() + ", "
-                        + "audio count: " + streamInfo.getAudioStreams().size() + ", "
-                        + "video count: " + streamInfo.getVideoOnlyStreams().size() + ", "
-                        + streamInfo.getVideoStreams().size();
-                return (ManagedMediaSource)
-                        FailedMediaSource.of(stream, new MediaSourceResolutionException(message));
-            }
-
-            final MediaItemTag tag = MediaItemTag.from(source.getMediaItem()).get();
-            final long expiration = System.currentTimeMillis()
-                    + ServiceHelper.getCacheExpirationMillis(streamInfo.getServiceId());
-            return new LoadedMediaSource(source, tag, stream, expiration);
+            final var source = playbackListener.sourceOf(stream, streamInfo);
+
+            return Optional.ofNullable(source)
```

Comment:
And also turn `source1` into `source` below

```suggestion
            return Optional.ofNullable(playbackListener.sourceOf(stream, streamInfo))
```